tests/testthat/test_ExtractData_Soils.R

```diff
@@ -1,268 +1,282 @@
 context("Soil data extraction")
 
-skip_on_travis()
-skip_on_appveyor()
-skip_on_cran()
-# whether or not these tests should be run
-run_tests <- FALSE
+# skip_on_travis() and skip_on_appveyor() are meant to be used within
+# test_that() calls. Here, we need to skip preparation code outside of a
+# test_that call. However, the skip_* functions cause CIs to error out, even
+# when wrapped in try() statements, with
+#   - "Error: On Appveyor", respectively
+#   - "Error: On Travis"
+# Check values of the ENV variables directly as a work-around:
 
-# =============================================================================
-# Tests designed to test the underlining structures created
-# from soil extraction functions.
-# =============================================================================
+do_skip <- c(
+  !identical(tolower(Sys.getenv("RSFSW2_ALLTESTS")), "true"),
+  # whereas skip_on_cran() skips if not "true", I believe it should skip only
+  # if "false" (i.e., not "" and not "true")
+  identical(tolower(Sys.getenv("NOT_CRAN")), "false"),
+  # mimmics skip_on_travis():
+  identical(tolower(Sys.getenv("TRAVIS")), "true"),
+  # mimmics skip_on_appveyor():
+  identical(tolower(Sys.getenv("APPVEYOR")), "true"))
 
-# set stage manually so that future changes won't cause the test to fail ======
-# setup file paths
-fnames_in <- environment()
-fnames_in$fslayers <- file.path("YOUR_PATH/SWRuns_InputData_SoilLayers_v9.csv")
-fnames_in$fsoils <- "YOUR_PATH/SWRuns_InputData_soils_v12.csv"
-dir_ex_soil <- "/YOUR_PATH_SOILS_DATA/"
-resume <- TRUE
-verbose <- FALSE
-MMC <- environment()
-sim_size <- environment()
-sim_space <- environment()
-SFSW2_glovars <- environment()
-SFSW2_glovars[["slyrs_ids"]] <- seq.int(1, 20, 1)
-SFSW2_glovars[["slyrs_maxN"]] <- 20
-SWRunInformation <- data.frame(row.names <- c(1, 2, 3, 4, 5, 6))
-SWRunInformation[, 1] <- c(-106.2995, -106.2748, -106.2813, -106.2875,
-                          -106.2875, -106.2875)
-SWRunInformation[, 2] <- c(35.7655, 35.76451, 35.77990, 35.79530, 35.79530,
-                          35.79530)
-SWRunInformation[, 3] <- rep("ISRIC_SoilGrids250m", 6)
-SWRunInformation[, 4] <- rep.int(1, 6)
-colnames(SWRunInformation) <- c("X_WGS84", "Y_WGS84", "SoilTexture_source",
-                                "Include_YN_SoilSources")
-sim_size$expN <- 2
-sim_size$runsN_master <- 6
-sim_size$runIDs_sites <- c(1, 2, 3, 5, 6)
-sim_size$runsN_sites <- 5
-sim_size$runsN_total <- 12
-sim_size$runIDs_total <- c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12)
-sim_size$runsN_job <- 10
-sim_size$runsN_Pid <- 12
-sim_size$runsIDs_todo <- NULL
-sim_size$runsN_todo <- 0
-sim_size$digitsN_total <- 3
-sim_size$runIDs_sites_by_dbW <- c(1, 2, 3, 4, 5)
+suppressWarnings(is_online <-
```

Comment:
Why do you test whether this has online access? I don't see anything in your tests that requires online access.

Reply:
You asked me to model setting up whether or not to skip the tests on other tests, thus I removed the `run_tests` flag since no other test does it that way but I do agree that it was useful. The code being tested will tell you if the paths are wrong ie. if you run the tests with an incorrect soils extraction path if will tell you the folder or file does not exist. Are you suggesting that I check those in the tests before running them as well?

Reply:
- I only referred to the use of `skip* functions` which `are intended for use within test_that() blocks`, i.e., they don't skip if outside a `test_that` block as you had it
- I didn't suggest at all that you test online access and I didn't say anything about removing `run_tests`

- If you don't skip your tests if the external files don't exist, then running `devtools::test()` will fail -- except on your specific machine. I think this is bad. You are introducing here a new type of tests to our package -- your tests are the first to assume data outside the source package and besides online accessible data. Thus, you need to handle your new test gracefully and differently such that other people's workflow is not being broken.

Reply:
I understand the intended use of the skip functions but it sounds much more helpful to give a user a reason for skipping the test using skip() outside of `testthat` functions rather than giving no explanation when skipping by simply bypassing the code as is the case in tests such as `test_net_CDF_function.R`.

Reply:
> skipping the test using skip() outside of testthat functions

This doesn't work. Have you ever tried it out???

> give a user a reason for skipping the test ... rather than giving no explanation

No one prevented you from providing a reason for skipping; it would be easy enough to add a `if (!any(do_skip)) {} else {test_that("block name", {skip(your message)})}. Plus, a package user never sees the tests; it is only developers that see the tests; thus, the code for `do_skip` calculation in `test_netCDF_functions.R` provides a detailed explanation in the comments.